src/test/java/co/rsk/bitcoinj/script/ScriptTest.java

```diff
@@ -150,109 +135,121 @@ public void testMultiSig() {
+            mainnetParams);
+        Script recipientScript = ScriptBuilder.createOutputScript(userAddress);
+        Coin amountToSend = fundingTxOutput.getValue();
+        spendTx.addOutput(amountToSend, recipientScript);
+
         Sha256Hash sighash = spendTx.hashForSignature(0, multisigScript, SigHash.ALL, false);
-        BtcECKey.ECDSASignature party1Signature = key1.sign(sighash);
-        BtcECKey.ECDSASignature party2Signature = key2.sign(sighash);
-        TransactionSignature party1TransactionSignature = new TransactionSignature(party1Signature, SigHash.ALL, false);
-        TransactionSignature party2TransactionSignature = new TransactionSignature(party2Signature, SigHash.ALL, false);
+
+        List<TransactionSignature> signatures = new ArrayList<>();
+        for (int i = 0; i < expectedNumberOfRequiredSignatures; i++) {
+            BtcECKey.ECDSASignature ecdsaSignature = federationKeys.get(i).sign(sighash);
+            TransactionSignature txSignature = new TransactionSignature(ecdsaSignature, SigHash.ALL, false);
+            signatures.add(txSignature);
+        }
 
         // Create p2sh multisig input script
-        Script inputScript = ScriptBuilder.createP2SHMultiSigInputScript(ImmutableList.of(party1TransactionSignature, party2TransactionSignature), multisigScript);
+        Script p2shScript = ScriptBuilder.createP2SHMultiSigInputScript(signatures, multisigScript);
 
-        // Assert that the input script contains 4 chunks
-        assertEquals(4, inputScript.getChunks().size());
+        // Assert that the input script contains standard multisig op codes + expected number of required signatures
+        int expectedNumberOfChunksForScriptSig = redeemScriptStandardOpCodesNumberOfChunks + expectedNumberOfRequiredSignatures;
+        assertEquals(expectedNumberOfChunksForScriptSig, p2shScript.getChunks().size());
 
         // Assert that the input script created contains the original multisig
         // script as the last chunk
-        ScriptChunk scriptChunk = inputScript.getChunks().get(inputScript.getChunks().size() - 1);
-        Assert.assertArrayEquals(scriptChunk.data, multisigScript.getProgram());
+        ScriptChunk redeemScriptChunk = p2shScript.getChunks().get(p2shScript.getChunks().size() - 1);
+        Assert.assertArrayEquals(redeemScriptChunk.data, multisigScript.getProgram());
 
         // Create regular multisig input script
-        inputScript = ScriptBuilder.createMultiSigInputScript(ImmutableList.of(party1TransactionSignature, party2TransactionSignature));
+        Script multiSigInputScript = ScriptBuilder.createMultiSigInputScript(signatures);
 
-        // Assert that the input script only contains 3 chunks
-        assertEquals(3, inputScript.getChunks().size());
+        // Assert that the input script only contains the OP_CODE zero and the required number of signatures.
+        int expectedOpCodesNumberOfChunks = 1;
```

Comment:
Shouldn't it be final and upper case?

Reply:
Why upper case?

Reply:
it is a constant, isn't it?

Reply:
That name sounds confusing to me at least. Can't understand what it means

Reply:
This is the number of `OP_0` opcodes expected in a multi-sig script. Renamed the variable to `expectedOpZeroNumberOfChunks`.